Update Mixpanel JS to version 2 API and add support for set_config options.

jamiequint opened the pull request:
All API requests with the new JS code are asynchronous so just removed that option for JS based requests (although should just be ignored if the option is there so this should remain backwards compatible)

jamiequint committed:
update to mixpanel 2 api

jamiequint committed:
add support for set_config options, update readme

jamiequint committed:
integrate people api

warmwaffles commented:
I would like to see this implemented

tomdev commented:
Just though about implementing this, but is already done. Would like to see this in the main branch.

jamiequint commented:
I'd like to note that I did not implement the 'increment' method of the API yet: https://mixpanel.com/docs/people-analytics/javascript#increment and may not as I'm no longer using this gem due to issue with it properly tracking requests in certain scenarios, should be trivial if someone want's to fork and tackle that though. I'm also not handling all special properties here but that should be trivial to update as well they are as follows (could not find them listed in the API docs but talked to Mixpanel support and this is what they told me):

"$email"
"$first_name"
"$last_name"
"$created"
"$last_login"

warmwaffles commented:
@jamiequint you also need to update the version number in the Gemspec

kelso committed:
fixed special property name in people api

kelso committed:
fixed mixpanel initialization - added missing quotes

jamiequint committed:
Merge pull request #1 from kelso/master

Fixed People API

RSO committed:
Added support for all special_properties in people api

RSO committed:
Created support for people.increment

RSO committed:
Cleaned up middleware_spec and added tests for people methods

RSO committed:
Removed debug statement

RSO committed:
Removed debugger statement

RSO committed:
Added more special_properties

Source:
https://mixpanel.com/docs/people-analytics/people-http-specification-ins
ert-data

hammerdr commented:
Any update on this? We would like to upgrade ourselves.

warmwaffles commented:
I have been using it in our production environment with ease

`gem 'mixpanel', :git => 'git://github.com/zippykid/mixpanel.git'`

Thought I would like to see it merged in to the main stream

jamiequint committed:
Merge pull request #2 from Factlink/mixpanel_people_api

Mixpanel people api

Commit:
Added persisting functionality

Commit:
Added queue merging

AlainPilon commented:
+1 for the merge in master

rschmukler committed:
Updated documentation for persistence layer

rschmukler committed:
Gave myself a little credit

rschmukler committed:
Fixed persistence documentation

jamiequint committed:
Merge pull request #3 from rschmukler/master

Added Persistence layer

zevarito commented:
For some reason I never get notified about this, I'll check it right now.

zevarito commented:
Guys, I don't have much time to look commit by commit, I've merged the full PR since you agree that was a good idea but it breaks the test suite, It is in master now, but will not be published until test suite becomes green again.

Can anyone take a look at it? Does the test suite pass for everyone?

zevarito commented:
Also if one of you want to become maintainer of the gem, please tell me.

warmwaffles commented:
I wouldn't mind maintaining it but, I wouldn't be contributing but I'd be able to facilitate the pull requests and doing code review

zevarito commented:
@warmwaffles that would work.

Beside tests failing I noted some api changes that aren't reflect on documentation and also will break current users of the gem. I hope to get time today to revert the last merge and go through each commit.

zevarito commented:
@rschmukler seems like persistance feature has no tests at all, I am considering to revert those commits.

Can you tell me exactly what it is? Thanks!

Also I don't like how it offset initializer params, it will break other people gem usage without reason, can you fix it?

rschmukler commented:
@zevarito Yes, sorry, I didn't write tests. (Shame on me). The persist feature is described in the updated README.rdoc file, but basically rather than using a Rack env variable, it uses the Rack session to store the queue. It then doesn't render the queue on Redirects, and it removes duplicate events allowing a before_filter to be used even through redirects.

zevarito commented:
@rschmukler thanks for the update.

Yesterday I was able to fix the failing tests, not happy to introduce features without testing those first, but since all you are using that new code I think we can do a little exception this time :P

_Important_ New version will introduce api changes, initializers no longer will take options as arguments, all those will be passed through options hash. I'll update the docs accordingly.

zevarito commented:
@all FYI I've pushed latest version to rubygems.

http://rubygems.org/gems/mixpanel

You may want to look at updated docs, initializer now takes a Hash as options parameter instead of async=true, persist=true etc.
